src/System.Private.DataContractSerialization/src/System/Runtime/Serialization/AccessorBuilder.cs

```diff
@@ -0,0 +1,153 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using System.Runtime.Serialization;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace System.Runtime.Serialization
+{
+    internal static class FastInvokerBuilder
+    {
+        public delegate void Setter(ref object obj, object value);
+        public delegate object Getter(object obj);
+
+        private delegate void StructSetDelegate<T, T1>(ref T obj, T1 value);
+
+        private static MethodInfo s_buildGetAccessorInternal = typeof(FastInvokerBuilder).GetMethod(nameof(BuildGetAccessorInternal), BindingFlags.NonPublic | BindingFlags.Public | BindingFlags.Static);
+        private static MethodInfo s_buildSetAccessorInternal = typeof(FastInvokerBuilder).GetMethod(nameof(BuildSetAccessorInternal), BindingFlags.NonPublic | BindingFlags.Public | BindingFlags.Static);
+        private static MethodInfo s_make = typeof(FastInvokerBuilder).GetMethod(nameof(Make), BindingFlags.NonPublic | BindingFlags.Public | BindingFlags.Static);
+
+        public static Func<object> GetMakeNewInstanceFunc(Type type)
+        {
+            Func<object> make = (Func<object>)s_make.MakeGenericMethod(type).CreateDelegate(typeof(Func<object>));
+            return make;
+        }
+
+        public static Getter BuildGetAccessor(MemberInfo memberInfo)
```

Comment:
Thanks, good suggestions. Can I make such changes later? I have another fix depending on the changes. I'm afraid renaming these would make rebasing the other fix onto the top of this harder.